### PR TITLE
AV-178913 : This commit adds documentation for ICAP Profile reference in HostRule…

### PR DESCRIPTION
**Pull request by saurabhvaidvmware**
… CRD

### CODE REVIEW

File: docs/crds/hostrule.md [hunk 1]
```diff
@@ -122,6 +124,16 @@ prior to this CRD creation. The application profile should be of `TYPE` of `APPL
  This property can be applied only for secure FQDNs and cannot be applied for insecure routes.
  The application profiles can be used for various HTTP/HTTP2 protocol settings.
 
+#### Express custom ICAP profile
+
+HostRule CRD can be used to express an ICAP profile reference. The ICAP profile reference should have been created in the Avi Controller prior to this CRD creation.
+
+        icapProfile:
```
Comment (swathinsankaran) [inline docs/crds/hostrule.md:131]: Better to mention that we will be considering only one icap profile reference from the list of references.

File: docs/crds/hostrule.md [hunk 2]
```diff
@@ -122,6 +124,16 @@ prior to this CRD creation. The application profile should be of `TYPE` of `APPL
...
 
+#### Express custom ICAP profile
+
+HostRule CRD can be used to express an ICAP profile reference. The ICAP profile reference should have been created in the Avi Controller prior to this CRD creation.
+
+        icapProfile: 
+        - avi-icap-ref
+ 
+ This property can be applied for both secure and insecure hosts via EVH parent and child Virtual Services, SNI child Virtual Services and dedicated VS's.
+ The ICAP profile can be used for transporting HTTP traffic to 3rd party services for processes such as content sanitization and antivirus scanning.
```
Comment (swathinsankaran) [inline docs/crds/hostrule.md:135]: Please provide the link to configure the ICAP profile in the controller if any as a reference here.

File: docs/crds/hostrule.md (hunk 1 above)
Comment (saurabhvaidvmware) [reply docs/crds/hostrule.md:131]: I've changed it to mention explicitly that a single reference per host is allowed

File: docs/crds/hostrule.md (hunk 2 above)
Comment (saurabhvaidvmware) [reply docs/crds/hostrule.md]: Linked to the AVI documentation page for ICAP which has this info

**Comment by swathinsankaran** [comment]
build ako

**Review by swathinsankaran — APPROVED** [review]
LGTM

**Review by akshayhavile — APPROVED** [review]

**Pull request merged**